### PR TITLE
GeoPackage cannot write boolean fields [GEOT-5483][GEOS-7654]

### PR DESCRIPTION
**Pull request by ianturton**
Some work to make online tests work for GeoPackage. 
Includes a fix for GEOT-5483 - GeoPackage cannot write boolean fields
Adds GeoPkgBooleanOnlineTest* implementations. 
Adds a Boolean <-> Integer binding to allow reading of booleans.

### CODE REVIEW

**Comment by aaime** [comment]
It's a bit hard to say what happened to GeoPackage, there is a lot of reformats

**Comment by bencaradocdavies** [comment]
Jenkins CI has been having some problems so I restarted this build.

**Comment by bencaradocdavies** [comment]
@ianturton looks like a genuine compile failure this time (Travis CI silliness has relented):
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project gt-geopkg: Compilation failure: Compilation failure:
[ERROR] /home/travis/build/geotools/geotools/modules/unsupported/geopkg/src/test/java/org/geotools/geopkg/GeoPkgFeatureSourceOnlineTest.java:[57,24] error: featureSource is not public in JDBCFeatureSourceOnlineTest; cannot be accessed from outside package
[ERROR] /home/travis/build/geotools/geotools/modules/unsupported/geopkg/src/test/java/org/geotools/geopkg/GeoPkgFeatureSourceOnlineTest.java:[58,24] error: featureSource is not public in JDBCFeatureSourceOnlineTest; cannot be accessed
```

File: modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java [hunk 1]
```diff
@@ -358,71 +346,62 @@ else if (connPool instanceof ManageableDataSource) {
...
+        } catch (IOException ex) {
             throw new SQLException("Unable to add default spatial references.", ex);
         }
     }
 
-    public void addCRS(int srid, String srsName, String organization, int organizationCoordSysId, String definition, String description)
-            throws IOException {
+    public static void addCRS(Connection cx, int srid, String srsName, String organization,
+            int organizationCoordSysId, String definition, String description) throws IOException {
+        // try {
```
Comment (aaime) [inline modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java]: Leftover commented out section (there are others around here)

File: modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java [hunk 2]
```diff
@@ -436,7 +415,20 @@ public void addCRS(int srid, String srsName, String organization, int organizati
      * 
      */
     public void addCRS(CoordinateReferenceSystem crs, String auth, int srid) throws IOException {
-        this.addCRS(srid, auth + ":" + srid, auth, srid, crs.toWKT(), auth + ":" + srid);
+
+        Connection cx;
+        try {
```
Comment (aaime) [inline modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java]: No need to change right now, but a suggestion for the future: connection is AutoCloseable, a single try-with-resources could have been used here.

File: modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java [hunk 3]
```diff
@@ -1868,20 +1791,24 @@ static void runSQL(String sql, Connection cx) throws SQLException {
...
         }
     }
 
-    void runScript(String filename, Connection cx) throws SQLException{        
-        SqlUtil.runScript(getClass().getResourceAsStream(filename), cx);        
+    static void runScript(String filename, Connection cx) throws SQLException {
+        try {
+            SqlUtil.runScript(new GeoPackage().getClass().getResourceAsStream(filename), cx);
+        } catch (IOException e) {
+
```
Comment (aaime) [inline modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java]: Eeek, eatin away exceptions?

File: modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java [hunk 4]
```diff
@@ -72,7 +74,11 @@ public GeoPkgDialect(JDBCDataStore dataStore) {
 
     @Override
     public void initializeConnection(Connection cx) throws SQLException {
-        new GeoPackage(dataStore.getDataSource()).init(cx);
+        /*DataSource dataSource = dataStore.getDataSource();
```
Comment (aaime) [inline modules/unsupported/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java]: Commented out code section

**Comment by aaime** [comment]
Thanks Ian, merging!

**Comment by aaime** [comment]
@ianturton can you close the tickets that all these commits solved?

**Pull request merged**